internal/pool/pool.go

```diff
@@ -235,7 +239,7 @@ func (p *ConnPool) Get(ctx context.Context) (*Conn, error) {
 		return nil, err
 	}
 
-	for {
+	for i := 0; i < maxBadConnRetries; i++ {
```

Comment:
I guess we need some limit here after https://github.com/go-redis/redis/pull/1824 was merged, but this won't help with https://github.com/go-redis/redis/issues/1737.

Reply:
And I see that you already realize this :)